src/drive/mobile/containers/StackErrorBoundary.jsx

```diff
@@ -0,0 +1,95 @@
+/* global __TARGET__ */
+import React, { Component } from 'react'
+import ReactMarkdown from 'react-markdown'
+import tosIcon from '../assets/icons/icon-tos.svg'
+import { Modal, Icon, Button, translate } from 'cozy-ui/react'
+import styles from '../styles/tosupdated'
+
+const TosUpdatedModal = translate()(({ t, newTosLink, onAccept, onRefuse }) => (
+  <Modal closable={false}>
+    <Modal.ModalHeader />
+    <Modal.ModalDescription className={styles['tosupdated']}>
+      <Icon icon={tosIcon} width={96} height={96} />
+      <h2 className={styles['tosupdated-title']}>{t('TOS.updated.title')}</h2>
+      <ReactMarkdown
+        className={styles['tosupdated-desc']}
+        source={t('TOS.updated.detail', { link: newTosLink })}
+      />
+      <Button
+        extension="full"
+        label={t('TOS.updated.cta')}
+        onClick={onAccept}
+      />
+      <Button
+        subtle
+        size="small"
+        extension="full"
+        label={t('TOS.updated.disconnect')}
+        onClick={onRefuse}
+      />
+    </Modal.ModalDescription>
+  </Modal>
+))
+
+class StackErrorBoundary extends Component {
+  state = {
+    warnings: []
+  }
+
+  componentDidMount() {
+    if (__TARGET__ === 'mobile') {
+      this.checkIfUserActionIsRequired()
+      document.addEventListener('resume', this.checkIfUserActionIsRequired)
+    }
+  }
+
+  componentWillUnmount() {
+    if (__TARGET__ === 'mobile') {
+      document.removeEventListener('resume', this.checkIfUserActionIsRequired)
+    }
+  }
+
+  checkIfUserActionIsRequired = async () => {
+    const { client, router } = this.context
+    try {
+      await client.getOrCreateStackClient().fetch('GET', '/apps')
+      const wasBlocked = this.state.warnings.length !== 0
+      if (wasBlocked) {
+        this.setState({ warnings: [] })
+        router.push({ pathname: '/' })
+      }
+    } catch (e) {
+      if (e.status === 402) {
+        this.setState({ warnings: e.reason })
```

Comment:
Is `e.reason` an array here? This isn't very clear.

Reply:
yep, the stack returns an array in its response, that gets put into the `FetchError` `reason` property by `cozy-client`.